pkg/relay/register.go

```diff
@@ -102,7 +106,14 @@ SendPayloads:
 			fc.SentVerificationInc()
 		}
 	}
-	return <-fc.ExitCh
+
+	err := <-fc.ExitCh
```

Comment:
Is this expected to always receive? If so I'll add a comment.

Reply:
yes, otherwise nothing would ever finish :)